crates/rpc/rpc/src/eth/api/server.rs

```diff
@@ -178,13 +179,107 @@ where
         Err(internal_rpc_err("unimplemented"))
     }
 
+    // FeeHistory is calculated based on lazy evaluation of fees for historical blocks, and further
+    // caching of it in the LRU cache.
+    // When new RPC call is executed, the cache gets locked, we check it for the historical fees
+    // according to the requested block range, and fill any cache misses (in both RPC response
+    // and cache itself) with the actual data queried from the database.
+    // To minimize the number of database seeks required to query the missing data, we calculate the
+    // first non-cached block number and last non-cached block number. After that, we query this
+    // range of consecutive blocks from the database.
```

Comment:
and since we insert them afterwards, this is fine so even if we'r missing a block(range) in the cache it is filled after the db read?

Reply:
yep, correct